src/revwalk.rs

```diff
@@ -186,7 +217,11 @@ impl<'repo> Iterator for Revwalk<'repo> {
         };
         unsafe {
             try_call_iter!(raw::git_revwalk_next(&mut out, self.raw()));
-            Some(Ok(Binding::from_raw(&out as *const _)))
+            Some(if crate::panic::panicked() {
```

Comment:
I think here this will want to propagate the panic rather than convert that to an error